elastic/kibana
# [Metric threshold] Save the ECS group by fields at the AAD root level

Related to #183220

## Summary

This PR extracts `getEcsGroups` to a package to save ECS groups in the Alert As Data (AAD) document for the metric threshold rule.

### 🧪 How to test
- Create a metric threshold rule with multiple groups (both ECS and non-ECS fields)
- Check the related AAD document; you should be able to see the ECS fields at the root level and not see non-ECS fields there
- Check the same information for the recovered alerts
- Rules without group by should work as before
